aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsSpanMetricsProcessor.java

```diff
@@ -116,6 +125,57 @@ private void recordErrorOrFault(ReadableSpan span, Attributes attributes) {
     }
   }
 
+  /**
+   * Attempt to pull status code from spans produced by AWS SDK instrumentation (both v1 and v2).
+   * AWS SDK instrumentation does not populate http.status_code when non-200 status codes are
+   * returned, as the AWS SDK throws exceptions rather than returning responses with status codes.
+   * To work around this, we are attempting to get the exception out of the events, then calling
+   * getStatusCode (for AWS SDK V1) and statusCode (for AWS SDK V2) to get the status code fromt the
+   * exception. We rely on reflection here because we cannot cast the throwable to
+   * AmazonServiceExceptions (V1) or AwsServiceExceptions (V2) because the throwable comes from a
+   * separate class loader and attempts to cast will fail with ClassCastException.
+   *
+   * <p>TODO: Short term workaround. This can be completely removed once
+   * https://github.com/open-telemetry/opentelemetry-java-contrib/issues/919 is resolved.
+   */
+  @Nullable
+  private static Long getAwsStatusCode(SpanData spanData) {
+    InstrumentationScopeInfo scopeInfo = spanData.getInstrumentationScopeInfo();
+    if (scopeInfo == null) {
```

Comment:
This is unnecessary as spanData.getInstrumentationScopeInfo() is not annotated with `@Nullable` and does not return null.